internal/cli/atlas/data_lake/pipelines/create.go

```diff
@@ -0,0 +1,225 @@
+
+package pipelines
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/file"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas/mongodbatlasv2"
+)
+
+type CreateOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	store        store.PipelinesCreator
+	pipelineName string
+
+	sinkType             string
+	sinkMetadataProvider string
+	sinkMetadataRegion   string
+	sinkPartitionField   []string
+	sourceType           string
+	sourceClusterName    string
+	sourceCollectionName string
+	sourceDatabaseName   string
+	transform            []string
+	filename             string
+	fs                   afero.Fs
+}
+
+var ErrSourceTypeInvalid = errors.New("--sourceType invalid")
+var ErrSinkPartitionFieldInvalid = errors.New("--sinkPartitionField format invalid")
+var ErrSinkTransformInvalid = errors.New("--transform format invalid")
+
+func (opts *CreateOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var createTemplate = `Pipeline {{ .Name }} created.`
+
+func (opts *CreateOpts) Run() error {
+	createRequest, err := opts.newCreateRequest()
+	if err != nil {
+		return err
+	}
+
+	r, err := opts.store.CreatePipeline(opts.ConfigProjectID(), *createRequest)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+func isInt(s string) bool {
+	_, err := strconv.ParseInt(s, 10, 32)
+	return err == nil
+}
+
+func (opts *CreateOpts) validate() error {
+	if strings.ToUpper(opts.sourceType) != periodicCPS || strings.ToUpper(opts.sourceType) != onDemandCPS {
+		return fmt.Errorf("%w: expected either '%s' or '%s' got '%s'", ErrSourceTypeInvalid, periodicCPS, onDemandCPS, opts.sourceType)
+	}
+
+	for _, entry := range opts.sinkPartitionField {
+		entries := strings.Split(entry, ":")
+		if len(entries) != 2 || len(entries[0]) == 0 || len(entries[1]) == 0 || !isInt(entries[1]) {
+			return fmt.Errorf("%w: expected format is 'field:order' got '%s'", ErrSinkPartitionFieldInvalid, entry)
+		}
+	}
+
+	for _, entry := range opts.transform {
+		entries := strings.Split(entry, ":")
+		if len(entries) != 2 || len(entries[0]) == 0 || len(entries[1]) == 0 {
+			return fmt.Errorf("%w: expected format is 'type:field' got '%s'", ErrSinkTransformInvalid, entry)
+		}
+	}
+
+	return nil
+}
+
+func (opts *CreateOpts) newCreateRequest() (*atlasv2.IngestionPipeline, error) {
+	if len(opts.filename) > 0 {
```

Comment:
```suggestion
	if opts.filename != "" {
```
should be equivalent and simpler